src/components/dialog/dialog.js

```diff
@@ -114,6 +117,20 @@ const VUIDialog = ({
         dialog.setAttribute('tabindex', 1);
         dialog.setAttribute('aria-hidden', false);
 
+        // Alerts and modal dialog types don't have close buttons
+        if (hasCloseBtn && !isAlert && !isModal) {
+            const closeBtnEl = createEl({
+                element: 'button',
+                className: ['js-dialog-btn-close', closeButtonClassName],
```

Comment:
I think you said something about the click event being attached to the JS hook but I'm struggling to see where that's happening?